cmake/warnings.cmake

```diff
@@ -50,9 +50,7 @@ add_warning(all)
 add_warning(extra)
 add_warning(pedantic)
 add_warning(error) # treat all warnings as errors
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
This warning is a no op for clang: 
https://clang.llvm.org/docs/DiagnosticsReference.html#wstrict-overflow